addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -548,13 +567,13 @@ def action_confirm(self):
 
     def action_validate(self):
         current_employee = self.env.user.employee_id
-        for holiday in self:
-            if holiday.state != 'confirm':
-                raise UserError(_('Allocation request must be confirmed in order to approve it.'))
+        if any(holiday.state != 'confirm' for holiday in self):
+            raise UserError(_('Allocation request must be confirmed in order to approve it.'))
 
-            holiday.write({'state': 'validate'})
-            holiday.write({'approver_id': current_employee.id})
+        self.write({'state': 'validate'})
```

Comment:
Linting: you could concatenate the two write into a single call, makes it easier to read.

Reply:
Indeed, not sure what I was thinking